### PR TITLE
[react] Upgrade Typescript 4

### PR DESCRIPTION
**Pull request by qmonmert**
Upgrade **Typescript 4** for **React**

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/react/package.json.ejs [hunk 1]
```diff
@@ -169,7 +169,7 @@ limitations under the License.
     <%_ if (cypressTests) { _%>
     "cypress": "VERSION_MANAGED_BY_CLIENT_COMMON",
     <%_ } _%>
-    "typescript": "3.8.3",
+    "typescript": "4.0.2",
```
Comment (DanielFran) [inline generators/client/templates/react/package.json.ejs:172]: @qmonmert Maybe we should start using the package.json file and define here VERSION_MANAGED_BY_CLIENT_REACT until we update it also to vue and put it on VERSION_MANAGED_BY_CLIENT_COMMON?

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/client/templates/react/package.json.ejs (hunk 1 above)
Comment (mshima) [inline generators/client/templates/react/package.json.ejs:172]: Can you move to package.json and set:
```suggestion
    "typescript": "VERSION_MANAGED_BY_CLIENT_REACT",
```

Comment (mshima) [reply generators/client/templates/react/package.json.ejs:172]: You were faster. =)

Comment (qmonmert) [reply generators/client/templates/react/package.json.ejs]: done

Comment (qmonmert) [reply generators/client/templates/react/package.json.ejs]: thanks guys, done ✅

**Pull request merged**